### PR TITLE
Demoted deps as tools based on modified arch

### PR DESCRIPTION
**Pull request by stephanwlee**

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by yahoocla** [comment]
CLA is valid!

File: packages/mendel-pipeline/src/pipeline.js [hunk 1]
```diff
@@ -19,15 +19,16 @@ function MendelPipeline(options) {
...
     const transformer = new Transformer(options.transforms, options);
+    const depsResolver = new DepResolver(options);
 
-    // Pipeline steps
-    const initializer = new Initialize({registry, transformer}, options);
-    const watcher = new Watcher({registry, transformer}, options);
-    const reader = new Reader({registry, transformer}, options);
-    const ist = new IST({registry, transformer}, options);
-    const depsResolver = new DepResolver({registry, transformer}, options);
+    const toolset = {registry, transformer, depsResolver};
```
Comment (irae) [inline packages/mendel-pipeline/src/pipeline.js]: You really did it? if deps will be an accessory, why don't deps run inside the IST/GST workers instead? we would save the AST part, right?

Just asking out of curiosity, really.

**Comment by irae** [comment]
👍

File: packages/mendel-pipeline/src/pipeline.js (hunk 1 above)
Comment (stephanwlee) [inline packages/mendel-pipeline/src/pipeline.js]: If we impose a restriction that every plugin has to operate on AST (potentially one type?) and output AST, then yes. If not, no.

**Pull request merged**